index.ts

```diff
@@ -20,38 +20,47 @@ export interface EntryInfo {
   basename: string;
 }
 export type PathOrDirent = Dirent | Path;
-export type Tester = (path: EntryInfo) => boolean;
+export type Tester = (entryInfo: EntryInfo) => boolean;
 export type Predicate = string[] | string | Tester;
-
-function defaultOptions() {
-  return {
-    root: '.',
-    fileFilter: (_path: EntryInfo) => true,
-    directoryFilter: (_path: EntryInfo) => true,
-    type: FILE_TYPE,
-    lstat: false,
-    depth: 2147483648,
-    alwaysStat: false,
-    highWaterMark: 4096,
-  };
+export enum EntryTypes {
+  FILE_TYPE = 'files',
+  DIR_TYPE = 'directories',
+  FILE_DIR_TYPE = 'files_directories',
+  EVERYTHING_TYPE = 'all',
 }
+export type ReaddirpOptions = {
+  root: string;
+  fileFilter?: Predicate;
+  directoryFilter?: Predicate;
+  type?: EntryTypes | 'files' | 'directories' | 'files_directories' | 'all';
+  lstat?: boolean;
+  depth?: number;
+  alwaysStat?: boolean;
+  highWaterMark?: number;
+};
 
-export type ReaddirpOptions = ReturnType<typeof defaultOptions>;
+const defaultOptions: ReaddirpOptions = {
+  root: '.',
+  fileFilter: (_entryInfo: EntryInfo) => true,
+  directoryFilter: (_entryInfo: EntryInfo) => true,
+  type: EntryTypes.FILE_TYPE,
+  lstat: false,
+  depth: 2147483648,
+  alwaysStat: false,
+  highWaterMark: 4096,
+};
+Object.freeze(defaultOptions)
 
 const RECURSIVE_ERROR_CODE = 'READDIRP_RECURSIVE_ERROR';
 const NORMAL_FLOW_ERRORS = new Set(['ENOENT', 'EPERM', 'EACCES', 'ELOOP', RECURSIVE_ERROR_CODE]);
-const FILE_TYPE = 'files';
-const DIR_TYPE = 'directories';
-const FILE_DIR_TYPE = 'files_directories';
-const EVERYTHING_TYPE = 'all';
-const ALL_TYPES = [FILE_TYPE, DIR_TYPE, FILE_DIR_TYPE, EVERYTHING_TYPE];
-const DIR_TYPES = new Set([DIR_TYPE, FILE_DIR_TYPE, EVERYTHING_TYPE]);
-const FILE_TYPES = new Set([FILE_TYPE, FILE_DIR_TYPE, EVERYTHING_TYPE]);
+const ALL_TYPES = [EntryTypes.DIR_TYPE, EntryTypes.EVERYTHING_TYPE, EntryTypes.FILE_DIR_TYPE, EntryTypes.FILE_TYPE, 'files', 'directories', 'files_directories', 'all'];
```

Comment:
this is incorrect

Reply:
what exactly?

Reply:
you're duplicating same entries. `EntryTypes.DIR_TYPE, ..., 'directories'`

Reply:
yes, because i need to do it for not doing breaking changes. i could hard type the array as "string[]" if you like it more.

Reply:
done

Reply:
it would, because you arrays and sets were not typed.. so it will be typed as "EntryTypes". so you need ot pass EntryTypes to the method, which would have led to an breaking change, but now it is fine that way